server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/SeriesDataRandomReader.java

```diff
@@ -80,20 +84,44 @@
   private long currentPageEndTime = Long.MAX_VALUE;
 
 
-  public AbstractSeriesReader(Path seriesPath, TSDataType dataType, QueryContext context,
-      List<TsFileResource> seqFiles, List<TsFileResource> unseqFiles) {
+  public SeriesDataRandomReader(Path seriesPath, TSDataType dataType, QueryContext context,
+      QueryDataSource dataSource, Filter timeFilter, Filter valueFilter) {
     this.seriesPath = seriesPath;
     this.dataType = dataType;
     this.context = context;
-    this.seqFileResource = seqFiles;
-    this.unseqFileResource = sortUnSeqFileResources(unseqFiles);
+    this.seqFileResource = dataSource.getSeqResources();
+    this.unseqFileResource = sortUnSeqFileResources(dataSource.getUnseqResources());
+    this.timeFilter = timeFilter;
+    this.valueFilter = valueFilter;
   }
 
-  protected abstract Filter getFilter();
+  public SeriesDataRandomReader(Path seriesPath, TSDataType dataType, QueryContext context,
+      List<TsFileResource> seqFileResource, List<TsFileResource> unseqFileResource,
+      Filter timeFilter, Filter valueFilter) {
+    this.seriesPath = seriesPath;
+    this.dataType = dataType;
+    this.context = context;
+    this.seqFileResource = seqFileResource;
+    this.unseqFileResource = sortUnSeqFileResources(unseqFileResource);
+    this.timeFilter = timeFilter;
+    this.valueFilter = valueFilter;
+  }
 
   protected boolean satisfyFilter(Statistics statistics) {
-    return getFilter() == null
-        || getFilter().containStartEndTime(statistics.getStartTime(), statistics.getEndTime());
+    return timeFilter == null
+        || timeFilter.containStartEndTime(statistics.getStartTime(), statistics.getEndTime());
+  }
+
+  @Override
+  public boolean canUseCurrentChunkStatistics() {
```

Comment:
add javadoc, only be used for aggregate without value filter

Reply:
done